error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -491,7 +491,10 @@ Flux<Integer> before(int value) {
 
     @BeforeTemplate
     Flux<T> before(T value) {
-      return Mono.just(value).repeat().take(1);
+      return Refaster.anyOf(
+          Mono.just(value).repeat().take(1),
+          Flux.fromIterable(ImmutableList.of(value)),
```

Comment:
Good for thought: Couldn't we also include varargs for the `ImmutableList` case? Essentially avoiding every unnecessary list creation.

Reply:
Yes; good one! @Stephan202 mentioned this could extend to a `Matcher` that checks for singleton lists, and potentially even more as you suggest :D